jerry-core/ecma/builtin-objects/ecma-builtin-object-prototype.c

```diff
@@ -292,12 +268,7 @@ ecma_builtin_object_prototype_dispatch_routine (uint16_t builtin_routine_id, /**
 
     else
     {
-#if ENABLED (JERRY_ESNEXT)
-      ecma_value_t this_parameter = this_arg;
-#else /* !ENABLED (JERRY_ESNEXT) */
-      ecma_value_t this_parameter = ecma_make_object_value (obj_p);
-#endif /* ENABLED (JERRY_ESNEXT) */
-      ret_value = ecma_builtin_object_prototype_object_to_locale_string (this_parameter, obj_p);
+      ret_value = ecma_builtin_object_prototype_object_to_locale_string (this_arg);
```

Comment:
At this point the is already a ToObject call done on the "this_arg". The "..to_locale_string" method should be moved from here to ensure that the "..to_object" call is not done (line 270/246)